src/Parser/Strings/StringParser.cpp

```diff
@@ -154,8 +156,37 @@ size_t StringParser::parseVariable() {
       continue;
     }
 
+    if (c == ':') {
+      mods = parseModifiers();
+      continue;
+    }
+
     unexpectedCharacter(c, "a variable");
   }
 
   unexpectedEndOfInput();
 }
+
+std::vector<std::string> StringParser::parseModifiers() {
+  auto& t = tokenizer();
+  std::vector<std::string> mods = std::vector<std::string>();
+  std::string mod;
```

Comment:
You should use [`std::stringstream`](https://en.cppreference.com/w/cpp/header/sstream) over a string.